src/models/backbones/swin.py

```diff
@@ -0,0 +1,451 @@
+                    self.patch_embed.grid_size[1] // (2 ** i_layer)),
+                depth=depths[i_layer],
+                num_heads=num_heads[i_layer],
+                window_size=window_size,
+                mlp_ratio=mlp_ratio,
+                qkv_bias=qkv_bias,
+                drop=drop_rate, attn_drop=attn_drop_rate,
+                drop_path=dpr[sum(depths[:i_layer]):sum(depths[:i_layer + 1])],
+                norm_layer=norm_layer,
+                downsample=PatchMerging if (i_layer < self.num_layers - 1) else None,
+                pretrained_window_size=pretrained_window_sizes[i_layer]
+            )
+            self.layers.append(layer)
+
+        self.feature_norms = nn.ModuleList([norm_layer(chs) for i, chs in enumerate(self.encoder_channels)])
+
+        self.apply(self._init_weights)
+        for bly in self.layers:
+            bly._init_respostnorm()
+
+
+    def _init_weights(self, m):
+        if isinstance(m, nn.Linear):
+            trunc_normal_(m.weight, std=.02)
+            if isinstance(m, nn.Linear) and m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+
+    @torch.jit.ignore
+    def no_weight_decay(self):
+        nod = {'absolute_pos_embed'}
+        for n, m in self.named_modules():
+            if any([kw in n for kw in ("cpb_mlp", "logit_scale", 'relative_position_bias_table')]):
+                nod.add(n)
+        return nod
+
+    @torch.jit.ignore
+    def group_matcher(self, coarse=False):
+        return dict(
+            stem=r'^absolute_pos_embed|patch_embed',  # stem and embed
+            blocks=r'^layers\.(\d+)' if coarse else [
+                (r'^layers\.(\d+).downsample', (0,)),
+                (r'^layers\.(\d+)\.\w+\.(\d+)', None),
+                (r'^norm', (99999,)),
+            ]
+        )
+
+    @torch.jit.ignore
+    def set_grad_checkpointing(self, enable=True):
+        for l in self.layers:
+            l.grad_checkpointing = enable
+
+    def __forward_patch_emb(self, x):
+        x = self.patch_embed(x)
+        if self.ape:
+            x = x + self.absolute_pos_embed
+        x = self.pos_drop(x)
+        return x
+
+    def __normalize_with_bhwc_reshape(self, x: torch.Tensor, leyer_number: int):
+        """Convert swin BLC shape to BHWC.
+
+        Args:
+            x: input tensor.
+            layer_number: number os SWin layer.
+        
+        Retruns:
+            x: tensor with shape BHWC.
+        """
+        # B L C
+        x = self.feature_norms[leyer_number](x)
+        H = self.input_resolutions[leyer_number][0]
+        W = self.input_resolutions[leyer_number][1]
+        C = self.encoder_channels[leyer_number]
+        # B H W C
+        x = x.view(-1, H, W, C).permute(0, 3, 1, 2).contiguous()
+        return x
+
+    def forward_features(self, x):
+        downsample_attn = self.__forward_patch_emb(x)
+        features = []
+        for i, layer in enumerate(self.layers):
+            downsample_attn, attn = layer(downsample_attn)
```

Comment:
Why is there only one output from this layer in the officical documentation?

Reply:
So in feature we add only attn, not downsample_attn